front/src/modules/timesStops/hooks/useComputedMargins.tsx

```diff
@@ -0,0 +1,73 @@
+import type { SimulationResponseSuccess } from 'applications/operationalStudies/types';
+import type { TrainScheduleResult } from 'common/api/generatedEditoastApi';
+import type { OperationalPointWithTimeAndSpeed } from 'modules/trainschedule/components/DriverTrainScheduleV2/types';
+import { interpolateValue } from 'modules/trainschedule/components/DriverTrainScheduleV2/utils';
+
+import { marginRegExValidation } from '../consts';
+import { formatDigitsAndUnit } from '../utils';
+
+let marginUnit = '%';
+
+const useComputedMargins = (
+  trainSimulation: SimulationResponseSuccess,
+  opPoint: OperationalPointWithTimeAndSpeed,
+  nextOpPoint: OperationalPointWithTimeAndSpeed,
+  selectedTrainSchedule: TrainScheduleResult,
+  pathId: string
+) => {
+  const theoreticalMarginIndex = selectedTrainSchedule?.margins?.boundaries?.findIndex(
+    (id) => id === pathId
+  );
+  if (!theoreticalMarginIndex) {
+    return ['', '', ''];
+  }
+  const theoreticalMargin = selectedTrainSchedule.margins!.values[theoreticalMarginIndex + 1];
+  const theoMarginRegexMatches = theoreticalMargin?.match(marginRegExValidation);
+  marginUnit = theoMarginRegexMatches?.[3] || marginUnit;
```

Comment:
Remove the let l9
```suggestion
  const marginUnit = theoMarginRegexMatches?.[3] || '%';
```

Reply:
`marginUnit` is kept between the function calls to have use the same unit as the previous column or row

Reply:
Ok my bad, I missed that, thanks !